exec/java-exec/src/main/java/org/apache/drill/exec/testing/ExecutionControlsInjector.java

```diff
@@ -81,8 +83,20 @@ public void injectPause(final ExecutionControls executionControls, final String
       executionControls.lookupPauseInjection(this, desc);
 
     if (pauseInjection != null) {
-      logger.debug("Pausing at {}", desc);
-      pauseInjection.pause();
+      long pauseDuration = pauseInjection.getMsPause();
+      if ( pauseDuration > 0L) {
```

Comment:
maybe nitpicky but I would use -1 (or any negative value) to distinguish between timed-bounded pause or not (since 0 is a valid wait time for CountDownLatch)

Reply:
Agreed, but the only purpose of that check is to indicate during debugging that a non-zero timed pause was injected. Hence, the check.